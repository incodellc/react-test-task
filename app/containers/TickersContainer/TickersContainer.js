import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {connection} from '../../services/';
import {startFetching } from '../../actions';
import TickerList from '../../components/TickerList/TickerList';


export class  TickersContainer extends React.Component {
    componentDidMount() {
        // eslint-disable-next-line no-shadow
        const { startFetching } = this.props;
        startFetching();
    }
    render() {
        const {tickers, loading, error} = this.props;
        return (
            <div>
                {error && <p className="row">ERROR!!!!!!!!!!!!!</p>}
                {loading ? <p className="row">Loading...</p> : null}
                {tickers && tickers.length &&  <TickerList tickers={tickers} />}
            </div>
        );
    }
}

const mapStateToProps = ({tickers, loading, error}) => {
    return {
        tickers,
        loading,
        error
    };
};
const mapDispatchToProps = (dispatch) => {
    return{
        startFetching: startFetching(dispatch, connection)
    };
};

TickersContainer.propTypes = {
    tickers: PropTypes.array.isRequired,
    loading: PropTypes.bool.isRequired,
    error: PropTypes.object,
    startFetching: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(TickersContainer);

