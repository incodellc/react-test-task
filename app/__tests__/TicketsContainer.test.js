import React from 'react';
import { shallow } from 'enzyme';
import { TickersContainer } from '../containers/TickersContainer/TickersContainer';

describe('Tickets Container test ', () => {
    const props = {
        tickers: [],
        loading: false,
        error: null,
        startFetching: () => {}
    };
    describe('Tickers Container initial', () => {
        const tickersContainer = shallow(<TickersContainer {...props} />);

        it('should renders properly', () => {
            expect(tickersContainer).toMatchSnapshot();
        });
    });

    describe('Tickers Container is loading', () => {
        const nextProps = {
            ...props,
            loading: true
        };

        const tickersContainer = shallow(<TickersContainer {...nextProps} />);

        it('should renders properly', () => {
            expect(tickersContainer).toMatchSnapshot();
        });

        it ('should render preloader', () => {
            expect(tickersContainer.find('p').text()).toEqual('Loading...');
        });
    });

    describe('Tickers Container is error', () => {
        const nextProps = {
            ...props,
            error: {error: 'Unexpected error'}
        };

        const tickersContainer = shallow(<TickersContainer {...nextProps} />);

        it('should renders properly', () => {
            expect(tickersContainer).toMatchSnapshot();
        });

        it ('should render errorIndicator', () => {
            expect(tickersContainer.find('p').text()).toEqual('ERROR!!!!!!!!!!!!!');
        });
    });

    describe('Tickers Container render < TickersList />', () => {
        const nextProps = {
            ...props,
            tickers: [1]
        };

        const tickersContainer = shallow(<TickersContainer {...nextProps} />);

        it('should renders properly', () => {
            expect(tickersContainer).toMatchSnapshot();
        });

        it ('should render <TickersList /> template', () => {
            expect(tickersContainer.find('TickerList')).toHaveLength(1);
        });
    });

    describe('Tickers Container initial in componentDidMount', () => {
        const mockFetchGetItems = jest.fn();
        const nextProps = {
            ...props,
            startFetching: mockFetchGetItems
        };

        const tickersContainer = shallow(<TickersContainer {...nextProps} />);

        it('should renders properly', () => {
            expect(tickersContainer).toMatchSnapshot();
        });

        it ('should dispatches the `startFetching()` method it receives from props', () => {
            expect(mockFetchGetItems).toHaveBeenCalledTimes(1);
        });
    });

});
