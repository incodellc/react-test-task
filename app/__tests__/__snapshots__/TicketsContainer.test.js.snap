// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tickets Container test  Tickers Container initial in componentDidMount should renders properly 1`] = `
<div>
  0
</div>
`;

exports[`Tickets Container test  Tickers Container initial should renders properly 1`] = `
<div>
  0
</div>
`;

exports[`Tickets Container test  Tickers Container is error should renders properly 1`] = `
<div>
  <p
    className="row"
  >
    ERROR!!!!!!!!!!!!!
  </p>
  0
</div>
`;

exports[`Tickets Container test  Tickers Container is loading should renders properly 1`] = `
<div>
  <p
    className="row"
  >
    Loading...
  </p>
  0
</div>
`;

exports[`Tickets Container test  Tickers Container render < TickersList /> should renders properly 1`] = `
<div>
  <TickerList
    tickers={
      Array [
        1,
      ]
    }
  />
</div>
`;
